libs/common/src/main/java/org/opensearch/common/io/InputStreamContainer.java

```diff
@@ -56,4 +60,11 @@ public long getContentLength() {
     public long getOffset() {
         return offset;
     }
+
+    /**
+     * @return metadata of the source content.
+     */
+    public Map<String, String> getMetadata() {
+        return metadata;
```

Comment:
No, we are not using this currently. This is introduced to provide support for async download flow...This will be used when we integrate the async download flow for blob downloads.

Reply:
How about we avoid adding it until it is needed?